crates/sui-framework/packages/sui-framework/sources/kiosk/kiosk.move

```diff
@@ -470,7 +501,9 @@ module sui::kiosk {
 
     /// Mutably borrow an item from the `Kiosk`.
     /// Item can be `borrow_mut`ed only if it's not `is_listed`.
-    public fun borrow_mut<T: key + store>(self: &mut Kiosk, cap: &KioskOwnerCap, id: ID): &mut T {
+    public fun borrow_mut<T: key + store>(
+        self: &mut Kiosk, cap: &KioskOwnerCap, id: ID
+    ): &mut T {
         assert!(object::id(self) == cap.for, ENotOwner);
         assert!(has_item(self, id), EItemNotFound);
```

Comment:
Don't we need Type check here?  (has_item_with_type)

Reply:
We don't. It's guaranteed by the `&mut T`